packages/mermaid/src/diagrams/gantt/parser/gantt.jison

```diff
@@ -77,24 +77,31 @@ that id.
 <click>[\s\n]           this.popState();
 <click>[^\s\n]*         return 'click';
 
-"gantt"                     return 'gantt';
-"dateFormat"\s[^#\n;]+      return 'dateFormat';
-"inclusiveEndDates"         return 'inclusiveEndDates';
-"topAxis"                   return 'topAxis';
-"axisFormat"\s[^#\n;]+      return 'axisFormat';
-"tickInterval"\s[^#\n;]+    return 'tickInterval';
-"includes"\s[^#\n;]+        return 'includes';
-"excludes"\s[^#\n;]+        return 'excludes';
-"todayMarker"\s[^\n;]+      return 'todayMarker';
-\d\d\d\d"-"\d\d"-"\d\d      return 'date';
-"title"\s[^#\n;]+           return 'title';
-"accDescription"\s[^#\n;]+  return 'accDescription'
-"section"\s[^#:\n;]+        return 'section';
-[^#:\n;]+                   return 'taskTxt';
-":"[^#\n;]+                 return 'taskData';
-":"                         return ':';
-<<EOF>>                     return 'EOF';
-.                           return 'INVALID';
+"gantt"                         return 'gantt';
+"dateFormat"\s[^#\n;]+          return 'dateFormat';
+"inclusiveEndDates"             return 'inclusiveEndDates';
+"topAxis"                       return 'topAxis';
+"axisFormat"\s[^#\n;]+          return 'axisFormat';
+"tickInterval"\s[^#\n;]+        return 'tickInterval';
+"includes"\s[^#\n;]+            return 'includes';
+"excludes"\s[^#\n;]+            return 'excludes';
+"todayMarker"\s[^\n;]+          return 'todayMarker';
+.*weekday\s+monday[^\n]*        return 'weekday_monday'
```

Comment:
```suggestion
.*weekday\s+monday[^\n]+        return 'weekday_monday'
```

The difference between `direction` of `flow` and `weekday` in `gantt` that it could have another attribute in the same line, for example

```
flowchart TB title ...
```

whereas in `gantt` it shouldn't:

```
gantt
weekday sunday title ...
```

Please update the other weekdays with it.

Reply:
Hey, I've tried this, but it fails the tests I setup for it. For some reason the tests will only accept `gantt\nweekday monday \n` and not `gantt\nweekday monday\n` (note the whitespace after monday). This is why I initially went with `*` over `+` because that made the thing work.

Reply:
Okay, fully removing `[^\n]+` makes the tests work again, not sure if this has any other consequences. It looks good from my perspective at least.

Reply:
Sorry I didn't notice that I commented it there instead of here :)

> I'm not an export in jison, but maybe removing it should resolve this?
> 
> ```diff
> -.*weekday\s+monday[^\n]*
> +.*weekday\s+monday
> ```
> https://github.com/mermaid-js/mermaid/pull/4634#discussion_r1264358447

I guess it should be fine.